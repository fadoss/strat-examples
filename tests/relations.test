algorithms/relations

srew (1, 10) (10, 1) (2, 20) using revert .
srew (1, 2) (2, 3) (5, 6) using trans .
srew (1, 2) (2, 3) (3, 5) (6, 7) using dominoes .
srew (1, 2) (2, 3) (3, 4) (3, 5) (5, 6) (7, 1) using remove(1) .
