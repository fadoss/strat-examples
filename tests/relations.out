==========================================
srewrite in RELATION-STRATS : (1,10) (2,20) (10,1) using revert .

Solution 1
rewrites: 15
result Relation: (1,10) (10,1) (20,2)

No more solutions.
rewrites: 15
==========================================
srewrite in RELATION-STRATS : (1,2) (2,3) (5,6) using trans .

Solution 1
rewrites: 25
result Relation: (1,2) (1,3) (2,3) (5,6)

No more solutions.
rewrites: 25
==========================================
srewrite in RELATION-STRATS : (1,2) (2,3) (3,5) (6,7) using dominoes .

Solution 1
rewrites: 16
result Relation: (1,5) (6,7)

No more solutions.
rewrites: 16
==========================================
srewrite in RELATION-STRATS : (1,2) (2,3) (3,4) (3,5) (5,6) (7,1) using remove(
    1) .

Solution 1
rewrites: 8
result Entry: (7,1)

No more solutions.
rewrites: 8
Bye.
